robot/Cumulus/resources/NPSP.robot

```diff
@@ -594,4 +594,104 @@ Verify No Errors Displayed on AM Page And Object Group
     Page Should Not Contain Locator     gift_entry.page_error
     View Field Mappings Of The Object   Account 1
     Page Should Not Contain Locator     npsp_settings.page_error    warning     ${object_group} : ${field} (${field}__c)
-    Wait For Locator Is Not Visible     npsp_settings.field_error   ${field}__c
\ No newline at end of file
+    Wait For Locator Is Not Visible     npsp_settings.field_error   ${field}__c
+
+Change Object Permissions
+    [Documentation]  Adds or removes the Create, Read, Edit and Delete permissions for the specified object on the specified permission set..
+    [Arguments]  ${action}  ${objectapiname}  ${permset}
+
+    ${ns} =  Get NPSP Namespace Prefix
+
+    ${removeobjperms} =  Catenate  SEPARATOR=\n
+    ...  ObjectPermissions objperm;
+    ...  objperm = [SELECT Id, PermissionsRead, PermissionsEdit, PermissionsCreate, PermissionsDelete FROM ObjectPermissions 
+    ...  WHERE parentId IN ( SELECT id FROM permissionset WHERE PermissionSet.Name = '${permset}') 
+    ...  AND SobjectType='${ns}${objectapiname}']; 
+    ...  objperm.PermissionsRead = false; 
+    ...  objperm.PermissionsEdit = false; 
+    ...  objperm.PermissionsCreate = false; 
+    ...  objperm.PermissionsDelete = false; 
+    ...  update objperm; 
+
+    ${addobjperms} =  Catenate  SEPARATOR=\n
+    ...  String permid = [SELECT id FROM permissionset WHERE PermissionSet.Name = '${permset}'].id;
+    ...  ObjectPermissions objperm = New ObjectPermissions(PermissionsRead = true, PermissionsEdit = true, PermissionsCreate = true, 
+    ...  PermissionsDelete = true, ParentId = permid, SobjectType='${ns}${objectapiname}');
+    ...  insert objperm;
+
+    Run Keyword if  "${action}" == "remove"
+    ...             Run Task  execute_anon
+    ...             apex= ${removeobjperms}
+
+    Run Keyword if  "${action}" == "add"
+    ...             Run Task  execute_anon
+    ...             apex= ${addobjperms}
```

Comment:
This is crazy ingenuity and I really love it. I think this could easily be turned into Python for reuse/performance.

Reply:
Good suggestion - I'll look into doing that.